src/features/click-to-load.js

```diff
@@ -1389,18 +1590,19 @@ export default class ClickToLoad extends ContentFeature {
         }
 
         // Listen for events from "surrogate" scripts.
-        addEventListener('ddg-ctp', (event) => {
+        addEventListener('ddg-ctp', (/** @type {CustomEvent} */ event) => {
             if (!event.detail) return
```

Comment:
I think this is not narrowing TS unfortunately, the following would remove the need for some of the ?'s below
```suggestion
            if (!('detail' in event)) return
```

Reply:
Done.